src/test/java/com/vaadin/flow/component/checkbox/tests/CheckboxGroupTest.java

```diff
@@ -163,4 +164,81 @@ public void changeItems_selectionIsReset() {
         Assert.assertThat(checkboxGroup.getValue(), IsEmptyCollection.empty());
         Assert.assertThat(capture.get(), IsEmptyCollection.empty());
     }
+
+    @Test
+    public void singleDataRefreshEvent() {
```

Comment:
Please add another test where e.g. `ListDataProvider` overrides the `getId(T item)` method and doesn't return the item itself but something else like an id property.